pkg/cmd/deploy/manifest.go

```diff
@@ -0,0 +1,378 @@
+	if err != nil {
+		return nil, err
+	}
+
+	manifest := Manifest{}
+	err = json.Unmarshal(b, &manifest)
+	if err != nil {
+		return nil, err
+	}
+
+	return &manifest, err
+}
+
+// Interpreted TODO: better interpreted, removed flows edge application, domain
+func (manifest *Manifest) Interpreted(f *cmdutil.Factory, cmd *DeployCmd, conf *contracts.AzionApplicationOptions, clients *Clients) error {
+	logger.Debug("Execute manifest")
+	ctx := context.Background()
+
+	err := cmd.doApplication(clients.EdgeApplication, ctx, conf)
+	if err != nil {
+		return err
+	}
+
+	domainName, err := cmd.doDomain(clients.Domain, ctx, conf)
+	if err != nil {
+		return err
+	}
+
+	err = cmd.doBucket(clients.Bucket, ctx, conf)
+	if err != nil {
+		return err
+	}
+
+	// skip upload when type = javascript (storage folder does not exist in this case)
+	if conf.Template != "javascript" {
+		err = cmd.uploadFiles(f, conf)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, route := range manifest.Routes {
+		if route.Type == "compute" {
+			conf.Function.File = ".edge/worker.js"
+			err := cmd.doFunction(clients.EdgeFunction, ctx, conf)
+			if err != nil {
+				return err
+			}
+
+			reqIns := apiEdgeApplications.CreateInstanceRequest{}
+			reqIns.SetEdgeFunctionId(conf.Function.ID)
+			reqIns.SetName(conf.Name)
+			reqIns.ApplicationId = conf.Application.ID
+			instance, err := clients.EdgeApplication.CreateInstancePublish(ctx, &reqIns)
+			if err != nil {
+				logger.Debug("Error while creating edge function instance", zap.Error(err))
+				return fmt.Errorf(msg.ErrorCreateInstance.Error(), err)
+			}
+			InstanceID = instance.GetId()
+		}
+
+		err = cmd.doOrigin(clients.EdgeApplication, clients.Origin, ctx, conf, route.Type)
+		if err != nil {
+			logger.Debug("Error while creating origin", zap.Error(err))
+			return err
+		}
+		logger.FInfo(cmd.F.IOStreams.Out, msg.OriginsSuccessful)
+
+		ruleDefaultID, err := clients.EdgeApplication.GetRulesDefault(ctx, conf.Application.ID, "request")
+		if err != nil {
+			logger.Debug("Error while updating rules engine", zap.Error(err))
+			return err
+		}
+
+		behaviors := make([]sdk.RulesEngineBehaviorEntry, 0)
+
+		var behString sdk.RulesEngineBehaviorString
+		behString.SetName("set_origin")
+
+		if route.Type == "compute" {
+			behString.SetTarget(strconv.Itoa(int(conf.Origin.SingleOriginID)))
+		} else {
+			behString.SetTarget(strconv.Itoa(int(conf.Origin.StorageOriginID)))
+		}
+
+		behaviors = append(behaviors, sdk.RulesEngineBehaviorEntry{
+			RulesEngineBehaviorString: &behString,
+		})
+
+		reqUpdateRulesEngine := apiEdgeApplications.UpdateRulesEngineRequest{
+			IdApplication: conf.Application.ID,
+			Phase:         "request",
+			Id:            ruleDefaultID,
+		}
+
+		reqUpdateRulesEngine.SetBehaviors(behaviors)
+
+		_, err = clients.EdgeApplication.UpdateRulesEngine(ctx, &reqUpdateRulesEngine)
+		if err != nil {
+			logger.Debug("Error while updating rules engine", zap.Error(err))
```

Comment:
in order to facilitate the debugging process, let's change this message to:
Error while updating default rules engine